src/client/debugger/extension/configuration/providers/fastapiLaunch.ts

```diff
@@ -0,0 +1,70 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+'use strict';
+
+import { inject, injectable } from 'inversify';
+import * as path from 'path';
+import { WorkspaceFolder } from 'vscode';
+import { IFileSystem } from '../../../../common/platform/types';
+import { DebugConfigStrings } from '../../../../common/utils/localize';
+import { MultiStepInput } from '../../../../common/utils/multiStepInput';
+import { sendTelemetryEvent } from '../../../../telemetry';
+import { EventName } from '../../../../telemetry/constants';
+import { DebuggerTypeName } from '../../../constants';
+import { LaunchRequestArguments } from '../../../types';
+import { DebugConfigurationState, DebugConfigurationType, IDebugConfigurationProvider } from '../../types';
+
+@injectable()
+export class FastAPILaunchDebugConfigurationProvider implements IDebugConfigurationProvider {
+    constructor(@inject(IFileSystem) private fs: IFileSystem) {}
+    public isSupported(debugConfigurationType: DebugConfigurationType): boolean {
+        return debugConfigurationType === DebugConfigurationType.launchFastAPI;
+    }
+    public async buildConfiguration(input: MultiStepInput<DebugConfigurationState>, state: DebugConfigurationState) {
+        const application = await this.getApplicationPath(state.folder);
+        let manuallyEnteredAValue: boolean | undefined;
+        const config: Partial<LaunchRequestArguments> = {
+            name: DebugConfigStrings.fastapi.snippet.name(),
+            type: DebuggerTypeName,
+            request: 'launch',
+            module: 'uvicorn',
+            args: ['main:app'],
+            jinja: true
+        };
+
+        if (!application) {
+            const selectedPath = await input.showInputBox({
+                title: DebugConfigStrings.fastapi.enterAppPathOrNamePath.title(),
+                value: 'main.py',
+                prompt: DebugConfigStrings.fastapi.enterAppPathOrNamePath.prompt(),
+                validate: (value) =>
+                    Promise.resolve(
+                        value && value.trim().length > 0
+                            ? undefined
+                            : DebugConfigStrings.fastapi.enterAppPathOrNamePath.invalid()
+                    )
+            });
+            if (selectedPath) {
+                manuallyEnteredAValue = true;
+                config.args = [`${selectedPath.replace('.py', '').replace('/', '.')}:app`];
```

Comment:
Since you already import `path`, an option here could be to use Node's [path.basename](https://nodejs.org/api/path.html#path_path_basename_path_ext):
```suggestion
                config.args = [`${path.basename(selectedPath, '.py').replace('/', '.')}:app`];
```

Reply:
done

Reply:
Thank you, that's what I wanted :)